docs/source/branch.rst

```diff
@@ -8,125 +8,167 @@ The Branches File
-simulation configurations are determined when both are specified. The specification of multiple branch configurations
-that would result in more than one simulation will lead to simulations for the cartesian product of the
-configuration sets. An example may make this clearer, so consider the following model specification.
+Since specifying either :term:`input draws<Input Draw>` or :term:`random seeds<Random Seed>` will result in multiple
+simulations being run, it is important to understand how :term:`branch configurations<Branch Configuration>` are
+parsed into simulations when both keys are specified. Specifying both an ``input_draw_count`` and a
+``random_seed_count`` will result in a set of input draws and a set of random seeds being independently
+generated. Simulations will then be run for each unique combination of input draw and random seed (the
+Cartesian product of the two sets).
+
+An example may make this clearer, so consider the following model specification.
 
 .. code-block:: yaml
+    :caption: combined_uncertainty_branches.yaml
 
     input_draw_count: 100
     random_seed_count: 10
 
 It combines the two configuration keys we just learned about. Taken separately, the ``input_draw_count`` mapping would
-lead to 100 simulations on 100 random draws of input data while the ``random_seed_count`` mapping would lead to ten
-simulations on ten randomly generated random seeds. With both specified, the result is one thousand total simulations,
-one for each member of the cartesian product of those sets. That is, ten simulations with the ten random seeds for each
-of the 100 GBD draws.
-
-Parameter Variations
---------------------
-A major function of branches files is to enable easy manipulation of the configuration parameters of a model
-specification to generate different scenarios or examine the sensitivity of a model to changes in a specific parameter.
-In the following sections we will describe a number of ways you can construct different scenarios and the effect on
-the number of simulations these will have.
+lead to 100 simulations on 100 draws of input data while the ``random_seed_count`` mapping would lead to ten
+simulations on with identical input data but a different seed for the random number generation. With both specified,
+the result is 1,000 total simulations, one for each member of the Cartesian product of those sets. That is,
+we would run ten simulations with the ten random seeds for each of the 100 input data draws.
+
+Configuration Variations
+------------------------
+
+A major function of :term:`branch configurations<Branch Configuration>` is to enable easy manipulation of
+the :term:`configuration parameters<Configuration Parameter>` of a :term:`model specification<Model Specification>`.
+These parameters generally govern interesting features of an intervention, such as it's target coverage or efficacy.
+
+Within a branch configuration, you can specify several variations of these parameters to generate different
```

Comment:
term reference for this branch configuration?

Reply:
I've been using the term reference any the first time a term shows up in a section.  I think even that may be a bit overzealous.